modules/audio_coding/neteq/packet_buffer.cc

```diff
@@ -99,8 +99,40 @@ PacketBuffer::~PacketBuffer() {
 
 // Flush the buffer. All packets in the buffer will be destroyed.
 void PacketBuffer::Flush(StatisticsCalculator* stats) {
+  // RingRTC change to log more information around audio jitter buffer flushes
+  auto prev_recv_ts = Timestamp::Micros(0);
+  auto num_out_of_order = 0;
+  auto num_gaps_below_15ms = 0;
+  auto num_gaps_above_40ms = 0;
+
   for (auto& p : buffer_) {
     LogPacketDiscarded(p.priority.codec_level, stats);
+    if (prev_recv_ts.us() > 0) {
+      auto gap_us = (p.packet_info.receive_time() - prev_recv_ts).us();
+
+      if (gap_us < 0) {
+        num_out_of_order++;
```

Comment:
Does this happen? I would expect the jitter buffer to keep the packets in order.

Reply:
It keeps the packets ordered by when they should be played out, but this is looking at the order that they're received in.

Reply:
/me fails today's Reading Comprehension test